src/device.cpp

```diff
@@ -88,6 +88,16 @@ void cvk_device::init_vulkan_properties(VkInstance instance) {
             "Failed to get pointer to vkGetPhysicalDeviceProperties2KHR()");
     }
     func(m_pdev, &properties);
+
+    //--- Get maxMemoryAllocationSize for figuring out the  max single buffer
+    // allocation size.
+    m_maintenance3_properties.sType =
+        VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_MAINTENANCE_3_PROPERTIES;
+    m_maintenance3_properties.pNext = nullptr;
+    m_properties2.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_PROPERTIES_2;
+    m_properties2.pNext = &m_maintenance3_properties;
+    vkGetPhysicalDeviceProperties2(m_pdev, &m_properties2);
```

Comment:
You don't need to call `vkGetPhysicalDeviceProperties2` here. This function is designed so that the queries are all chained in a single call. You can move the init line to the top of the function, along with the other struct initialisations and add an entry to `coreversion_extension_properties`. Everything else is automatic, including checking for the extension presence (I didn't spot that in my previous review but you should add the extension to the list of desired extensions in https://github.com/kpet/clvk/blob/main/src/device.cpp#L207).